### ISSUE TITLE
segmentation fault: reading 1KX5

### ISSUE THREAD

**Issue by hainm** [Amber-MD/cpptraj#122]
```bash
CPPTRAJ: Trajectory Analysis. V16.00b
    ___  ___  ___  ___
     | \/ | \/ | \/ | 
    _|_/\_|_/\_|_/\_|_

| Date/time: 10/13/15 19:03:01
| Available memory: 9365.43 MB

INPUT: Reading Input from file 1kx5.in
  [parm 1KX5.pdb]
	Reading '1KX5.pdb' as PDB File
Segmentation fault
```

http://www.rcsb.org/pdb/explore.do?structureId=1kx5

**Comment by hainm** [comment]
I tried with parmed and got better error message `FormatNotFound: Could not identify file format` :D

**Comment by swails** [comment]
I suspect the issue here is that the residue numbers in 1KX5 are negative.  At least that was why ParmEd could not autodetect this PDB file (fixed in the referenced PR).

Note that ParmEd parsed this PDB file just fine as long as you gave it directly to the PDB parser.  You can verify this with

```Python
import parmed as pmd

pmd.download_PDB('1kx5')
```

It wasn't immediately clear to me why cpptraj is segfaulting, but I suspect it has something to do with the negative residue numbers.

**Comment by drroe** [comment]
It's actually because CONECT records were being read improperly. Otherwise cpptraj seems to read the file in OK. I have a PR, #125, in the works that addresses this.

**Comment by hainm** [comment]
@swails 

I did `wget http://www.rcsb.org/pdb/files/1KX5.pdb` and `pmd.load_file('1KX5.pdb')`
what is the difference from `pmd.download_PDB('1kx5')`. I though both should call the PDB parser?

**Comment by swails** [comment]
`pmd.download_PDB` opens a URL and passes the open URL to `PDBFile.parse`.  `pmd.load_file` walks through every registered file format type, calls `FormatType.id_format(filename)`, and once that routine returns `True` for one of the types, it then dispatches that file to `FormatType.parse()`.

This PDB file was failing in `FormatType.id_format` (as indicated by "file format not found"), which is never called in `pmd.download_PDB`.

Likewise, you could have replaced `pmd.load_file('1KX5.pdb')` with `pmd.formats.PDBFile.parse('1KX5.pdb')` and it would've also worked.

**Comment by drroe** [comment]
Fixed by #125
